pkg/process/config/config.go

```diff
@@ -323,7 +324,7 @@ func NewAgentConfig(loggerName config.LoggerName, yamlPath, netYamlPath string)
 		cfg.LogLevel = "warn"
 	}
 
-	if cfg.HostName == "" {
+	if err := validate.ValidHostname(cfg.HostName); err != nil {
 		// lookup hostname if there is no config override
```

Comment:
nit: update this comment 
```suggestion
		// lookup hostname if there is no config override or if override is invalid
```